server/src/services/metadata.service.ts

```diff
@@ -478,6 +488,146 @@ export class MetadataService implements OnEvents {
     }
   }
 
+  async searchPerson(ownerId: AssetEntity['ownerId'], dto: SearchPeopleDto): Promise<PersonResponseDto[]> {
+    return this.personRepository.getByName(ownerId, dto.name, { withHidden: dto.withHidden });
+  }
+
+  private toPixels(region: MetadataRegion, unit: string): MetadataRegion {
+    const pixelRegion: MetadataRegion = region;
+    if (unit.toLowerCase() == 'normalized') {
+      return {
+        imageWidth: region.imageWidth,
+        imageHeight: region.imageHeight,
+        x1: Math.floor(region.x1 * region.imageWidth),
+        x2: Math.floor(region.x2 * region.imageWidth),
+        y1: Math.floor(region.y1 * region.imageHeight),
+        y2: Math.floor(region.y2 * region.imageHeight),
+      };
+    }
+    return pixelRegion;
+  }
+
+  private async applyTaggedFaces(asset: AssetEntity, tags: ImmichTags) {
+    this.logger.debug(`Extracting face metadata from (${asset.id})`);
+
+    if (!tags.RegionInfo) {
+      return true;
+    }
+
+    // Stores all face regions
+    const faceSet = new Set<MetadataFaceResult>();
+
+    // Stores all face names
+    const nameSet = new Set<string>();
+
+    // Stores AssetFace entities
+    const discoveredFaces: Partial<AssetFaceEntity>[] = [];
+
+    for (const region of tags.RegionInfo?.RegionList) {
+      const faceResultName = region.Name || 'unknown';
+      const faceResult: MetadataFaceResult = {
+        Name: faceResultName,
+        Type: region.Type,
+        Region: this.toPixels(
+          {
+            imageWidth: tags.RegionInfo?.AppliedToDimensions?.W,
+            imageHeight: tags.RegionInfo?.AppliedToDimensions?.H,
+            x1: region.Area.X - region.Area.W / 2,
+            y1: region.Area.Y - region.Area.H / 2,
+            x2: region.Area.X + region.Area.W / 2,
+            y2: region.Area.Y + region.Area.H / 2,
+          },
+          region.Area.Unit,
+        ),
+      };
+      faceSet.add(faceResult);
+      nameSet.add(faceResultName);
+    }
+    this.logger.debug(`${faceSet.size} faces detected in ${asset.originalPath}`);
+
+    if (nameSet.size === 0) {
+      return true;
+    }
+
+    // Find all person entries matching the face names we have
+    const matches = await this.personRepository.getManyByName(asset.ownerId, [...nameSet], { withHidden: true });
```

Comment:
You're doing `[...nameSet]` and `[...missing]` in multiple places. Prefer to do it once since this is rebuilding the array each time.